### ISSUE TITLE
AS3 3.27 return error when patch /Common/Shared 

### ISSUE THREAD

**Issue by BenEkpy** [F5Networks/f5-appsvcs-extension#447]
### Environment
 * Application Services Version: 3.27
 * BIG-IP Version: 15.1

### Summary
When adding a pool using the PATCH method in /Common/Shared, AS3 return an error when node are the same but stil create the pool.

### Steps To Reproduce

1. Define /Common/Shared with some pools using the same node 
```json
{
	"class": "ADC",
	"schemaVersion": "3.27.0",
	"updateMode": "selective",
	"Common": {
		"class": "Tenant",
		"Shared": {
			"class": "Application",
			"template": "shared",
			"pool-1": {
				"class": "Pool",
				"members": [
					{
						"servicePort": 80,
						"serverAddresses": [
							"1.1.1.1"
						]
					}
				]
			},
			"pool-2": {
				"class": "Pool",
				"members": [
					{
						"servicePort": 80,
						"serverAddresses": [
							"1.1.1.1"
						]
					}
				]
			}
		}
	}
}
```

2. Pools are created without errors
```json
{
  "results": [
    {
      "code": 200,
      "message": "success",
      "lineCount": 21,
      "host": "localhost",
      "tenant": "Common",
      "runTime": 3472
    },
    {
      "code": 200,
      "message": "no change",
      "host": "localhost",
      "tenant": "Common",
      "runTime": 1739
    }
  ],
  "declaration": {
    "class": "ADC",
    "schemaVersion": "3.27.0",
    "updateMode": "selective",
    "Common": {
      "class": "Tenant",
      "Shared": {
        "class": "Application",
        "template": "shared",
        "pool-1": {
          "class": "Pool",
          "members": [
            {
              "servicePort": 80,
              "serverAddresses": [
                "1.1.1.1"
              ]
            }
          ]
        },
        "pool-2": {
          "class": "Pool",
          "members": [
            {
              "servicePort": 80,
              "serverAddresses": [
                "1.1.1.1"
              ]
            }
          ]
        }
      }
    },
    "id": "autogen_cfb248a6-eac6-4550-8c33-7fbcc4418537",
    "controls": {
      "archiveTimestamp": "2021-04-23T14:24:06.433Z"
    }
  }
}
```

TMSH control : 

```
(/Common/Shared)(tmos)# list ltm pool one-line
ltm pool pool-1 { members { 1.1.1.1:http { address 1.1.1.1 metadata { source { value declaration } } } } min-active-members 1 }
ltm pool pool-2 { members { 1.1.1.1:http { address 1.1.1.1 metadata { source { value declaration } } } } min-active-members 1 }
```

3. Add a pool in /Common/Shared using patch method :
```json
{
	"class": "AS3",
	"action": "patch",
	"patchBody": [
		{
			"op": "add",
			"path": "/Common/Shared/pool_3",
			"value": {
				"class": "Pool",
				"members": [
					{
						"servicePort": 80,
						"serverAddresses": [
							"1.1.1.1"
						]
					}
				]
			}
		}
	]
}
```

3. AS3 return an error : 
```json
{
  "results": [
    {
      "code": 200,
      "message": "success",
      "lineCount": 17,
      "host": "localhost",
      "tenant": "Common",
      "runTime": 2805
    },
    {
      "code": 422,
      "message": "declaration failed",
      "response": "01070110:3: Node address '/Common/Shared/1.1.1.1' is referenced by a member of pool '/Common/Shared/pool-2'.",
      "host": "localhost",
      "tenant": "Common",
      "runTime": 3097
    }
  ],
  "declaration": {
    "Common": {
      "class": "Tenant",
      "Shared": {
        "class": "Application",
        "template": "shared",
        "pool-1": {
          "class": "Pool",
          "members": [
            {
              "servicePort": 80,
              "serverAddresses": [
                "1.1.1.1"
              ]
            }
          ]
        },
        "pool-2": {
          "class": "Pool",
          "members": [
            {
              "servicePort": 80,
              "serverAddresses": [
                "1.1.1.1"
              ]
            }
          ]
        }
      }
    },
    "class": "ADC",
    "schemaVersion": "3.27.0",
    "updateMode": "selective",
    "id": "autogen_0e256b87-22dd-4253-b6cb-e787f7a03f68",
    "controls": {
      "archiveTimestamp": "2021-04-23T14:31:47.129Z"
    }
  },
  "code": 207
}
```

TMSH control : 

```
(/Common/Shared)(tmos)# list ltm pool one-line
ltm pool pool-1 { members { 1.1.1.1:http { address 1.1.1.1 metadata { source { value declaration } } } } min-active-members 1 }
ltm pool pool-2 { members { 1.1.1.1:http { address 1.1.1.1 metadata { source { value declaration } } } } min-active-members 1 }
ltm pool pool_3 { members { 1.1.1.1:http { address 1.1.1.1 metadata { source { value declaration } } } } min-active-members 1 }
```

Even if AS3 return an error, pool is created.

### Expected Behavior

No error should be return in this case. 
Tests done in version < 3.27 are fines

**Comment by dstokesf5** [comment]
Thank you for reporting this. I have confirmed the bug and added it to our internal product backlog as AUTOTOOL-2457.

**Comment by dstokesf5** [comment]
It looks like this bug was resolved by some other work that was done for the 3.29.0 release.

**Comment by dstokesf5** [comment]
This issue has been resolved by the 3.29.0 release.
